src/lib/dnssd/minimal_mdns/Server.cpp

```diff
@@ -258,6 +258,7 @@ CHIP_ERROR ServerBase::Listen(chip::Inet::InetLayer * inetLayer, ListenIterator
             ChipLogError(DeviceLayer, "MDNS failed to join multicast group on %s for address type %s: %s", interfaceName,
                          AddressTypeStr(addressType), chip::ErrorStr(err));
             ShutdownEndpoint(mEndpoints[endpointIndex]);
+            continue;
```

Comment:
The continue should be explained

Reply:
This was found by Mingjie ... his pool change fixes it differently.

Generally if shutdown is called, we do not want to alloocate a UDP endpoint below (that will leak)